avocado/utils/network.py

```diff
@@ -45,36 +46,45 @@ def is_port_free(port, address):
     return free
 
 
-def find_free_port(start_port, end_port, address="localhost"):
+def find_free_port(start_port, end_port, address="localhost", sequent=True):
     """
     Return a host free port in the range [start_port, end_port].
 
-    :param start_port: First port that will be checked.
-    :param end_port: Port immediately after the last one that will be checked.
+    :param start_port: header of candidate port range
+    :param end_port: ender of candidate port range
+    :param sequent: Find port sequently, random order if it's False
     :param address: Socket address to bind or connect
     """
-    for i in range(start_port, end_port):
+    port_range = range(start_port, end_port)
+    if not sequent:
+        random.shuffle(port_range)
+    for i in port_range:
         if is_port_free(i, address):
             return i
     return None
 
 
-def find_free_ports(start_port, end_port, count, address="localhost"):
+def find_free_ports(start_port, end_port, count, address="localhost", sequent=True):
     """
     Return count of host free ports in the range [start_port, end_port].
 
-    :param start_port: First port that will be checked.
-    :param end_port: Port immediately after the last one that will be checked.
+    :param start_port: header of candidate port range
+    :param end_port: ender of candidate port range
     :param count: Initial number of ports known to be free in the range.
     :param address: Socket address to bind or connect
+    :param sequent: Find port sequently, random order if it's False
     """
     ports = []
-    i = start_port
-    while i < end_port and count > 0:
+
+    port_range = range(start_port, end_port)
+    if not sequent:
+       random.shuffle(port_range)
```

Comment:
This line is missing an extra space, causing a style verification failure.

Reply:
Sorry for that